ci/jenkins/README.md

```diff
@@ -1 +1,81 @@
 # Nephe CI: Jenkins
+
+## Table of Contents
+
+<!-- toc -->
+- [Introduction](#introduction)
+- [Jenkins Deployment](#jenkins-deployment)
+  - [Teraform](#teraform)
+  - [Ansible](#ansible)
+  - [Jenkins Job](#jenkins-job)
+    - [Requirements](#requirements)
+    - [Apply the jobs](#apply-the-jobs)
+    - [Jenkins job updater](#jenkins-job-updater)
+- [List of jenkins job](#list-of-jenkins-jobs)
+<!-- /toc -->
+
+## Introduction
+
+This directory includes all the scripts required to run CI for nephe.
+
+We have a cluster of static VMs configured to run as jenkins node.  When a jenkins
+job is triggered, the nephe code will be cloned on the static VM. The jenkins node
+will call terraform to deploy a dynamic VM and use ansible for installing any missing
+packages like docker. Then tests will be run on that dynamic VM, Upon completion
+of the test (success or fail), the dynamic VM will be destroyed.
+
+## Jenkins Deployment
+
+### Teraform
+
+The persistent data stored in the backend belongs to a workspace. Named workspaces allow
+conveniently switching between multiple instances of a single configuration within its single
+backend. Multiple workspaces are used here to create multiple dynamic VMs in parallel, each
+dynamic VM is in a workspace.
+
+-[nephi-ci.sh](./nephe-ci.sh) is responsible for creating terraform tfvars files.
```

Comment:
- [nephi-ci.sh](./nephe-ci.sh) is responsible for creating terraform tfvars files.
- [deploy.sh](./deploy.sh) is the script to create vm, and store the terraform state
   in a specified directory, which can be used to retrieve VM information.
- [destroy.sh](./destroy.sh) helps to destroy the dynamic VM and delete terraform workspace as well.

Reply:
done